### PR TITLE
[flutter_tools] remove globals/context for android testing

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Update AndroidDevice to accept all dependencies directly. Remove context and globals from android device testing where possible

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device_discovery.dart [hunk 1]
```diff
@@ -154,6 +162,12 @@ class AndroidDevices extends PollingDeviceDiscovery {
             productID: info['product'],
             modelID: info['model'] ?? deviceID,
             deviceCodeName: info['device'],
+            androidSdk: androidSdk ?? globals.androidSdk,
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_device_discovery.dart:165]: I don't love this, but I also didn't want to change absolute everything.

Medium term, we should split up the device object from the device service.

**Review by jmagman — APPROVED** [review]
LG(ross)TM

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 2]
```diff
@@ -117,16 +124,17 @@ void main() {
...
       expect(launchResult.started, true);
       expect(processManager.hasRemainingExpectations, false);
-    }, overrides: <Type, Generator>{
-      AndroidSdk: () => androidSdk,
-      FileSystem: () => fileSystem,
-      ProcessManager: () => processManager,
     });
   }
 
   testUsingContext('AndroidDevice.startApp does not allow release builds on x86', () async {
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart:130]: Can this be `testWithoutContext`?

**Review by zanderso — APPROVED** [review]

**Pull request merged**